store/src/java/com/zimbra/cs/volume/VolumeCLI.java

```diff
@@ -252,12 +287,103 @@ private void addVolume() throws ParseException, SoapFaultException, IOException,
         vol.setRootPath(path);
         vol.setCompressBlobs(compress != null ? Boolean.parseBoolean(compress) : false);
         vol.setCompressionThreshold(compressThreshold != null ? Long.parseLong(compressThreshold) : 4096L);
+        validateAddCommand(vol);
         CreateVolumeRequest req = new CreateVolumeRequest(vol);
         auth();
         CreateVolumeResponse resp = JaxbUtil.elementToJaxb(getTransport().invokeWithoutSession(
                 JaxbUtil.jaxbToElement(req)));
         System.out.println("Volume " + resp.getVolume().getId() + " is created");
     }
+    
+    /**
+     * This method validate the attributes in edit command.
+     * 
+     * @param volumeInfo, volStoreType
+     * @throws ParseException
+     */
+
+    private void validateEditCommand(VolumeInfo volumeInfo, Volume.StoreType volStoreType) throws ParseException {
+        if (volStoreType.equals(Volume.StoreType.INTERNAL)) {
+            if (!Strings.isNullOrEmpty(type)) {
+                volumeInfo.setType(toType(type));
+            }
+            if (!Strings.isNullOrEmpty(path)) {
+                volumeInfo.setRootPath(path);
+            }
+            if (!Strings.isNullOrEmpty(compress)) {
+                volumeInfo.setCompressBlobs(Boolean.parseBoolean(compress));
+            }
+            if (!Strings.isNullOrEmpty(compressThreshold)) {
+                volumeInfo.setCompressionThreshold(Long.parseLong(compressThreshold));
+            }
+        } else if (volStoreType.equals(Volume.StoreType.EXTERNAL)) {
+            if (!Strings.isNullOrEmpty(type)) {
```

Comment:
This is the mandatory parameter I think, regardless of store_type

Reply:
This is the edit command operation and this is mandatory only in add command. If we have store type external then we have to edit the name only.

Reply:
Yeah got it. thanks :)